pkg/virt-api/rest/subresource.go

```diff
@@ -401,6 +427,46 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 		return
 	}
 
+	// Only force restart is supported for now
+	// Check if bodyMap["graceperiod"] is not empty because gracePeriodint64 can be 0 if that is empty
+	if gracePeriodint64 == 0 && bodyMap["graceperiod"] != "" {
+		pods, err := app.virtCli.CoreV1().Pods(namespace).List(k8smetav1.ListOptions{})
```

Comment:
This makes looking up the pod pretty expensive. We should use an informer for that. We already have an informer factory in virt-api. Have a look at pkg/virt-api/api.go. You can get a pod informer out of it `factory.KubeVirtPod()`, which provides a cache for pods which are relevant for kubevirt.